submissions/DmytryjK/memory-pair-game/js/script.js

```diff
@@ -0,0 +1,212 @@
+  {
+    hero: 'slark',
+    srcHero: 'images/images_cards/slark.jpg'
+  },
+  {
+    hero: 'riki',
+    srcHero: 'images/images_cards/riki.jpg'
+  },
+  {
+    hero: 'pa',
+    srcHero: 'images/images_cards/pa.jpg'
+  },
+  {
+    hero: 'ember',
+    srcHero: 'images/images_cards/ember.jpg'
+  },
+  {
+    hero: 'lancer',
+    srcHero: 'images/images_cards/lancer.jpg'
+  },
+  {
+    hero: 'gyro',
+    srcHero: 'images/images_cards/gyro.jpg'
+  },
+  {
+    hero: 'necr',
+    srcHero: 'images/images_cards/necr.jpg'
+  },
+  {
+    hero: 'void',
+    srcHero: 'images/images_cards/void.jpg'
+  }
+];
+
+const srcBg = 'images/back/dota_back.svg',
+  winnerPopup = document.querySelector('.winMessage'),
+  mainCardsBlock = document.querySelector('.main__cards-block');
+let doubleElementsOfCards = [];
+
+let parrentCardBlock = document.createElement('ul');
+parrentCardBlock.classList.add('main__cards');
+
+createElementsOfCards();
+
+let currentListCards = document.querySelectorAll('.main__card');
+let activeCards = [],
+  numbersOfHideCards = 0,
+  timeoutActiveCards = null;
+
+const currentTime = document.querySelector('.timer-current span'),
+  bestTime = document.querySelector('.timer-best span');
+
+let minutes = 0,
+  seconds = 0,
+  bestResultsSheet = [],
+  timerInterval = null;
+
+const buttonToContinue = document.querySelector('.click-to-continue');
+
+setTimeout(showAllCards, 400);
+setTimeout(hideAllCards, 2000);
+
+if (localStorage.getItem('bestTime') !== null) {
+  bestTime.innerText = localStorage.getItem('bestTime');
+  bestResultsSheet.push(bestTime.innerText);
+}
+
+function onClickUserActions(e) {
+  const clickedTarget = e.target.parentNode;
+
+  if (clickedTarget.classList.contains('main__card')) {
+    clickedTarget.classList.add('active-card');
+    activeCards.push(clickedTarget);
+  }
+
+  if (activeCards.length >= 2) {
+
+    if (activeCards[0].getAttribute(["data-name"]) === activeCards[1].getAttribute(["data-name"]) &&
+      activeCards[0].getAttribute(["data-numbers"]) !== activeCards[1].getAttribute(["data-numbers"])) {
+        setTimeout(hideActiveCards, 600, activeCards);
+    } else {
+      setTimeout(closeActiveCards, 800, activeCards);
+    }
+
+    activeCards = [];
+  }
+}
+
+function restartGame() {
+  resetVariables();
+  resetListeners();
+  createElementsOfCards();
+  currentListCards = document.querySelectorAll('.main__card');
+  resetTimer();
+  setTimeout(showAllCards, 400);
+  setTimeout(hideAllCards, 2000);
+}
+
+function hideActiveCards(activeCards) {
+  activeCards.forEach(activeCard => {
+    activeCard.classList.add('hide-card');
+  });
+  numbersOfHideCards++;
+  checkOnVictory();
+}
+
+function closeActiveCards(activeCards) {
+  activeCards.forEach(activeCard => {
+    activeCard.classList.remove('active-card');
+  });
+  activeCards = [];
+}
+
+function showAllCards() {
+  currentListCards.forEach(card => {
+    card.classList.add('active-card');
+  });
+}
+
+function hideAllCards() {
```

Comment:
The user still can see more than 2 cards by clicking the rest of the cards. please set restrictions on that)
[scrnli_10_15_2022_11-15-08 AM.webm](https://user-images.githubusercontent.com/16196199/195976727-0e46e68c-fdde-4034-beb9-8ce367ee516c.webm)

Reply:
I know what you mean, but earlier I write about this, I decided not to limit the user to choose only 2 cards on the field, I think this is better for his user experience)
But I can take it all back